### PR TITLE
[Winback]  Apply Apple API for subscription cancellation

### LINKED ISSUE #2444: [Winback] Add confirmation screen

### LINKED ISSUE #2445: [Winback] Implement StorerKit API to present the Apple Subscription screen

### PR DESCRIPTION
**Pull request by danielebogo**
| 📘 Part of: #2435
|:---:|

Fixes #2444 
Fixes #2445 

This PR uses the `StorerKit.AppStore` API to present the subscriptions panel and handle those

## To test

- CI must be 🟢 
- Make sure you enabled `winback` FF
- Use a real device with a sandbox account
- Use a Plus/Patron account or a free account and purchase a Plan
- Got to your account, Cancel Subscription
- Tap on Continue to Cancellation
- On the new screen tap on Yes, Cancel my Subscription 
- You should see the Apple screen to manage the sunscription
- Tap on cancel the subscriptions
- Dismiss the view
- Now, the API to check the receipt may take some times. I need to investigate if we can get the new status quickly. If the state doesn't reload try to force reload your account till the new status updates

## Checklist

- [ ] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/66">7.78</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/CancelConfirmationViewModel.swift [hunk 1]
```diff
@@ -25,7 +26,29 @@ class CancelConfirmationViewModel: OnboardingModel {
         Analytics.track(.cancelConfirmationCancelButtonTapped)
 
         if FeatureFlag.winback.enabled {
-            //TODO: Add Apple API
+            Task {
+                guard let windowScene = await navigationController.view.window?.windowScene else {
```
Comment (SergioEstevao) [inline podcasts/CancelConfirmationViewModel.swift:29]: Could this code to be refactored to a separated function that could part of IAPHelper in order to better contain any StoreKit code?

**Review by SergioEstevao — CHANGES_REQUESTED** [review]
Functionality wise is working correctly, just asked for a code change.

File: podcasts/CancelConfirmationViewModel.swift (hunk 1 above)
Comment (danielebogo) [reply podcasts/CancelConfirmationViewModel.swift:29]: @SergioEstevao I part of it to the `IAPHandler`. All the part related to the groupID and StoreKit, so now everything is well decoupled

**Review by SergioEstevao — APPROVED** [review]
Thanks for addressing the suggestion! 
:shipit:

**Pull request merged**